edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.CloudProxy/DeviceScopeApiClient.cs

```diff
@@ -133,6 +159,61 @@ async Task<ScopeResult> GetIdentitiesInScope(Uri uri)
             }
         }
 
+        async Task<ScopeResult> GetIdentitiesInTargetScopeWithRetry(Uri uri, Option<string> continuationToken)
+        {
+            try
+            {
+                return await ExecuteWithRetry(
+                    () => this.GetIdentitiesInTargetScope(uri, continuationToken),
+                    Events.RetryingGetIdentities,
+                    this.retryStrategy);
+            }
+            catch (Exception e)
+            {
+                Events.ErrorInScopeResult(e);
+                throw;
+            }
+        }
+
+        async Task<ScopeResult> GetIdentitiesInTargetScope(Uri uri, Option<string> continuationLink)
+        {
+            HttpClient client = this.proxy
+                .Map(p => new HttpClient(new HttpClientHandler { Proxy = p }, disposeHandler: true))
+                .GetOrElse(() => new HttpClient());
+            using (var msg = new HttpRequestMessage(HttpMethod.Post, uri))
+            {
+                // Get the auth-chain for the target device
+                if (!this.AuthChainProvider.TryGetAuthChain(this.targetEdgeDeviceId, out string authChain))
```

Comment:
>f (!this.AuthChainProvider.TryGetAuthChain(this.targetEdgeDeviceId, out string authChain)) [](start = 17, length = 90)

From David: Consider accumulating the authchain through the nested iterator, as opposed to re-calculating it at runtime everytime we make a request.